server/src/main/java/io/druid/client/indexing/IndexingServiceClient.java

```diff
@@ -94,9 +148,47 @@ public void upgradeSegments(String dataSource, Interval interval)
     runQuery(new ClientConversionQuery(dataSource, interval));
   }
 
+  public Map<String, Object> getSegmentMetadata(String dataSource, List<Interval> intervals)
+  {
+    runQuery(new ClientSegmentMetadataQuery(dataSource, intervals));
+    return null;
+  }
+
+  public List<Map<String, Object>> getIncompleteTasks()
+  {
+    final InputStream queryResponse = runGetQuery(INCOMPLETE_TASKS);
+    try {
+      return jsonMapper.readValue(
+          queryResponse, new TypeReference<List<Map<String, Object>>>()
+          {
+          }
+      );
+    }
+    catch (IOException e) {
+      log.warn("Error parsing response for incompleteTasks query from Overlord [%s]", baseUrl());
+      return null;
+    }
+  }
+
+  private InputStream runGetQuery(String path)
+  {
+    try {
+      return client.go(
+          new Request(
+              HttpMethod.GET,
+              new URL(String.format("%s/%s", baseUrl(), path))
+          ), RESPONSE_HANDLER
+      ).get();
+    }
+    catch (Exception e) {
+      throw Throwables.propagate(e);
```

Comment:
Can you handle exceptions explicitly here?

Notably it is not clear if InterruptedExcepiton is handled in the way expected